pkg/sql/types/types.go

```diff
@@ -195,8 +195,13 @@ type UserDefinedTypeMetadata struct {
 	// this version of the type metadata.
 	Version uint32
 
-	// enumData is non-nil iff the metadata is for an ENUM type.
+	// EnumData is non-nil iff the metadata is for an ENUM type.
 	EnumData *EnumMetadata
+
+	// ImplicitRecordType is true if the metadata is for an implicit record type
+	// for a table. Note: this can be deleted if we migrate implicit record types
+	// to ordinary persisted composite types.
+	ImplicitRecordType bool
```

Comment:
Do we want this to be a table ID instead? Zero meaning it's not an implicit record type, and vice-versa.

Reply:
I don't see a use case for recording that table ID, so I guess not? :) We can easily add it if need be.